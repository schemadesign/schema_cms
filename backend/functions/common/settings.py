import os


HOST = os.getenv("LOCALSTACK_HOSTNAME", '')

AWS_STORAGE_BUCKET_NAME = os.getenv('AWS_STORAGE_BUCKET_NAME')
AWS_S3_ENDPOINT_URL = os.getenv(
    'AWS_S3_ENDPOINT_URL',
    f"http://{HOST}:4572"
)
AWS_ACCESS_KEY_ID = os.getenv('AWS_ACCESS_KEY_ID')
AWS_SECRET_ACCESS_KEY = os.getenv('AWS_SECRET_ACCESS_KEY')
AWS_DEFAULT_REGION = os.getenv('AWS_DEFAULT_REGION')
SECRET_MANAGER_ENDPOINT_URL = os.getenv(
    'SECRET_MANAGER_ENDPOINT_URL',
    f"http://{HOST}:4584"
)
LAMBDA_AUTH_TOKEN = os.getenv('LAMBDA_AUTH_TOKEN', '')

BACKEND_URL = os.getenv('BACKEND_URL', '')
