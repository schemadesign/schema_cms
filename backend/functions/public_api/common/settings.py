import json
import os


LOCAL_AWS_SERVICES_HOST = os.getenv("LOCALSTACK_HOSTNAME", None)

AWS_S3_ENDPOINT_URL = os.getenv("AWS_S3_ENDPOINT_URL")

if LOCAL_AWS_SERVICES_HOST:
    AWS_S3_ENDPOINT_URL = f"http://{LOCAL_AWS_SERVICES_HOST}:4572"


AWS_STORAGE_BUCKET_NAME = os.getenv("AWS_STORAGE_BUCKET_NAME")
IMAGE_SCRAPING_FETCH_TIMEOUT = int(os.getenv("IMAGE_SCRAPING_FETCH_TIMEOUT", 15))
AWS_IMAGE_STORAGE_BUCKET_NAME = os.getenv("AWS_IMAGE_STORAGE_BUCKET_NAME")
AWS_IMAGE_STATIC_URL = os.getenv("AWS_IMAGE_STATIC_URL")

AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID")
AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY")
AWS_DEFAULT_REGION = os.getenv("AWS_DEFAULT_REGION")

LAMBDA_AUTH_TOKEN = os.getenv("LAMBDA_AUTH_TOKEN", "")

BACKEND_URL = os.getenv("BACKEND_URL", "")

SENTRY_DNS = os.getenv("SENTRY_DNS", None)


def get_local_db():
    return {
        "password": os.getenv("POSTGRES_PASSWORD"),
        "dbname": os.getenv("POSTGRES_DB"),
        "port": os.getenv("POSTGRES_PORT"),
        "host": os.getenv("POSTGRES_HOST"),
        "username": os.getenv("POSTGRES_USER"),
    }


DB_CONNECTION = os.getenv("DB_CONNECTION", json.dumps(get_local_db()))
