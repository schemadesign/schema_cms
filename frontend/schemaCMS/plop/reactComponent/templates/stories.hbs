import React from 'react';
import { storiesOf } from '@storybook/react';

import { {{ pascalCase name }} } from './{{ camelCase name }}.component';

export const defaultProps = {};

storiesOf('{{ pascalCase name }}', module).add('Default', () => <{{ pascalCase name }} {...defaultProps} />);
