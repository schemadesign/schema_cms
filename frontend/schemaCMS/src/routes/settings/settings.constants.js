import { FormattedMessage } from 'react-intl';
import React from 'react';

import messages from './settings.messages';
import { LINK_ITEM } from '../../shared/components/menu/mobileMenu/mobileMenu.constants';
import { ROLES } from '../../modules/userProfile/userProfile.constants';

export const SETTINGS_PROJECTS_ID = 'projectsNavBtn';
export const SETTINGS_USERS_ID = 'usersNavBtn';

export const SETTINGS_MENU_OPTIONS = [
  {
    label: <FormattedMessage {...messages.projectsListPage} />,
    to: '/project/',
    id: SETTINGS_PROJECTS_ID,
    type: LINK_ITEM,
    allowedRoles: [ROLES.ADMIN, ROLES.EDITOR],
  },
  {
    label: <FormattedMessage {...messages.usersPage} />,
    to: '/user/',
    id: SETTINGS_USERS_ID,
    type: LINK_ITEM,
    allowedRoles: [ROLES.ADMIN],
  },
];
