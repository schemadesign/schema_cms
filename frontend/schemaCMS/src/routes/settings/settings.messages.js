/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'settings.title',
    defaultMessage: 'Schema CMS',
  },
  subTitle: {
    id: 'settings.subTitle',
    defaultMessage: 'My Profile',
  },
  save: {
    id: 'settings.save',
    defaultMessage: 'Save',
  },
  resetPassword: {
    id: 'settings.resetPassword',
    defaultMessage: 'Reset password',
  },
  projectsListPage: {
    id: 'settings.projectsListPage',
    defaultMessage: 'Projects',
  },
  usersPage: {
    id: 'settings.usersPage',
    defaultMessage: 'Users',
  },
});
