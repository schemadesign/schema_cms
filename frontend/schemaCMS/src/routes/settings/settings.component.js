import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Formik } from 'formik';
import { always } from 'ramda';

import { UserProfile } from '../../shared/components/userProfile/userProfile.component';
import { TopHeader } from '../../shared/components/topHeader';
import { ContextHeader } from '../../shared/components/contextHeader';
import messages from './settings.messages';
import {
  AUTH_METHODS,
  INITIAL_VALUES,
  USER_PROFILE_FORM,
  USER_PROFILE_SCHEMA,
} from '../../modules/userProfile/userProfile.constants';
import { Link, LinkContainer } from '../../theme/typography';
import { BackButton, NavigationContainer, NextButton } from '../../shared/components/navigation';
import { Form } from './settings.styles';
import { renderWhenTrue } from '../../shared/utils/rendering';
import { errorMessageParser } from '../../shared/utils/helpers';

export class Settings extends PureComponent {
  static propTypes = {
    updateMe: PropTypes.func.isRequired,
    userData: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
  };

  handleSubmit = async (values, { setSubmitting, setErrors }) => {
    try {
      setSubmitting(true);
      await this.props.updateMe(values);
    } catch (errors) {
      const errorMessages = errorMessageParser({ errors });
      setErrors(errorMessages);
    } finally {
      setSubmitting(false);
    }
  };

  handleBack = () => {
    this.props.history.goBack();
  };

  renderResetPasswordLink = renderWhenTrue(
    always(
      <LinkContainer>
        <Link onClick={() => this.props.history.push('/reset-password')}>
          <FormattedMessage {...messages.resetPassword} />
        </Link>
      </LinkContainer>
    )
  );

  renderContent = ({ dirty, handleSubmit, isSubmitting, ...restProps }) => {
    return (
      <Form onSubmit={handleSubmit}>
        <UserProfile {...restProps} isCurrentUser />
        {this.renderResetPasswordLink(this.props.userData.authMethod === AUTH_METHODS.EMAIL)}
        <NavigationContainer fixed>
          <BackButton type="button" onClick={this.handleBack} />
          <NextButton type="submit" loading={isSubmitting} disabled={!dirty || isSubmitting}>
            <FormattedMessage {...messages.save} />
          </NextButton>
        </NavigationContainer>
      </Form>
    );
  };

  render() {
    const headerTitle = <FormattedMessage {...messages.title} />;
    const headerSubtitle = <FormattedMessage {...messages.subTitle} />;

    return (
      <Fragment>
        <TopHeader headerTitle={headerTitle} headerSubtitle={headerSubtitle} />
        <ContextHeader title={headerTitle} subtitle={headerSubtitle} />
        <Formik
          enableReinitialize
          displayName={USER_PROFILE_FORM}
          validationSchema={USER_PROFILE_SCHEMA}
          onSubmit={this.handleSubmit}
          initialValues={{
            ...INITIAL_VALUES,
            ...this.props.userData,
          }}
          render={this.renderContent}
        />
      </Fragment>
    );
  }
}
