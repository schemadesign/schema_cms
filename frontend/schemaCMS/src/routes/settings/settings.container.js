import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';
import { bindPromiseCreators, promisifyRoutine } from 'redux-saga-routines';

import { Settings } from './settings.component';
import { selectUserData, UserProfileRoutines } from '../../modules/userProfile';
import { ProjectRoutines } from '../../modules/project';

const mapStateToProps = createStructuredSelector({
  userData: selectUserData,
});

export const mapDispatchToProps = dispatch => ({
  ...bindPromiseCreators(
    {
      updateMe: promisifyRoutine(UserProfileRoutines.updateMe),
      clearProject: promisifyRoutine(ProjectRoutines.clearProject),
    },
    dispatch
  ),
});

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(Settings);
