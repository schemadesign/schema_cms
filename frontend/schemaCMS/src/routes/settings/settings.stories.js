import React from 'react';
import { storiesOf } from '@storybook/react';

import { Settings } from './settings.component';
import { withTheme } from '../../.storybook/decorators';

export const defaultProps = {
  updateMe: Function.prototype,
  userData: {},
  history: { push: Function.prototype, goBack: Function.prototype },
};

storiesOf('Settings', module)
  .addDecorator(withTheme())
  .add('Default', () => <Settings {...defaultProps} />);
