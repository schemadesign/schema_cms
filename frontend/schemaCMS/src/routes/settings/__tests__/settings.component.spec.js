import React from 'react';
import { shallow } from 'enzyme';

import { Settings } from '../settings.component';
import { defaultProps } from '../settings.stories';

describe('Settings: Component', () => {
  const component = props => <Settings {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
