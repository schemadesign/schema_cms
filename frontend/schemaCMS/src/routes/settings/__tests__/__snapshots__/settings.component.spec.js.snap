// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings: Component should render correctly 1`] = `
<Fragment>
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="My Profile"
        id="settings.subTitle"
        values={Object {}}
      />
    }
    headerTitle={
      <FormattedMessage
        defaultMessage="SchemaCMS"
        id="settings.title"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="My Profile"
        id="settings.subTitle"
        values={Object {}}
      />
    }
    title={
      <FormattedMessage
        defaultMessage="SchemaCMS"
        id="settings.title"
        values={Object {}}
      />
    }
  />
  <Formik
    displayName="user_profile_form"
    enableReinitialize={true}
    initialValues={
      Object {
        "authMethod": "email",
        "email": "",
        "firstName": "",
        "lastName": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "email",
          "lastName",
          "firstName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
          "lastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  />
</Fragment>
`;

exports[`Settings: Component should render correctly 2`] = `
<Fragment>
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="My Profile"
        id="settings.subTitle"
        values={Object {}}
      />
    }
    headerTitle={
      <FormattedMessage
        defaultMessage="SchemaCMS"
        id="settings.title"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="My Profile"
        id="settings.subTitle"
        values={Object {}}
      />
    }
    title={
      <FormattedMessage
        defaultMessage="SchemaCMS"
        id="settings.title"
        values={Object {}}
      />
    }
  />
  <Formik
    displayName="user_profile_form"
    enableReinitialize={true}
    initialValues={
      Object {
        "authMethod": "email",
        "email": "",
        "firstName": "",
        "lastName": "",
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={
      ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "email",
          "lastName",
          "firstName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
          "lastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      }
    }
  />
</Fragment>
`;

exports[`Settings: Component should render correctly form 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "authMethod": "email",
        "email": "",
        "firstName": "",
        "lastName": "",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setError": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validate": undefined,
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "email",
          "lastName",
          "firstName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
          "lastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      },
      "values": Object {
        "authMethod": "email",
        "email": "",
        "firstName": "",
        "lastName": "",
      },
    }
  }
>
  <styled.form
    onSubmit={[Function]}
  >
    <UserProfile
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      initialValues={
        Object {
          "authMethod": "email",
          "email": "",
          "firstName": "",
          "lastName": "",
        }
      }
      isCurrentUser={true}
      isValid={false}
      isValidating={false}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "authMethod": "email",
          "email": "",
          "firstName": "",
          "lastName": "",
        }
      }
    />
    <styled.div>
      <Styled(Span)
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Reset password"
          id="settings.resetPassword"
          values={Object {}}
        />
      </Styled(Span)>
    </styled.div>
    <NavigationContainer
      fixed={true}
      hideOnDesktop={false}
      right={false}
    >
      <BackButton
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <NextButton
        disabled={true}
        loading={false}
        type="submit"
      >
        <FormattedMessage
          defaultMessage="Save"
          id="settings.save"
          values={Object {}}
        />
      </NextButton>
    </NavigationContainer>
  </styled.form>
</ContextProvider>
`;

exports[`Settings: Component should render form without reset password link 1`] = `
<ContextProvider
  value={
    Object {
      "dirty": false,
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "authMethod": "gmail",
        "email": "",
        "firstName": "",
        "lastName": "",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setError": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validate": undefined,
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": ObjectSchema {
        "_blacklist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "_conditions": Array [],
        "_defaultDefault": [Function],
        "_deps": Array [],
        "_excludedEdges": Array [],
        "_exclusive": Object {},
        "_mutate": undefined,
        "_nodes": Array [
          "email",
          "lastName",
          "firstName",
        ],
        "_options": Object {
          "abortEarly": true,
          "recursive": true,
        },
        "_type": "object",
        "_typeError": [Function],
        "_whitelist": RefSet {
          "list": Set {},
          "refs": Map {},
        },
        "fields": Object {
          "email": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "required": true,
              "undefined": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
            ],
          },
          "firstName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
          "lastName": StringSchema {
            "_blacklist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "_conditions": Array [],
            "_deps": Array [],
            "_exclusive": Object {
              "max": true,
              "min": true,
              "required": true,
              "trim": false,
            },
            "_mutate": undefined,
            "_options": Object {
              "abortEarly": true,
              "recursive": true,
            },
            "_type": "string",
            "_typeError": [Function],
            "_whitelist": RefSet {
              "list": Set {},
              "refs": Map {},
            },
            "tests": Array [
              [Function],
              [Function],
              [Function],
              [Function],
            ],
            "transforms": Array [
              [Function],
              [Function],
            ],
          },
        },
        "tests": Array [],
        "transforms": Array [
          [Function],
        ],
      },
      "values": Object {
        "authMethod": "gmail",
        "email": "",
        "firstName": "",
        "lastName": "",
      },
    }
  }
>
  <styled.form
    onSubmit={[Function]}
  >
    <UserProfile
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      initialValues={
        Object {
          "authMethod": "gmail",
          "email": "",
          "firstName": "",
          "lastName": "",
        }
      }
      isCurrentUser={true}
      isValid={false}
      isValidating={false}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "authMethod": "gmail",
          "email": "",
          "firstName": "",
          "lastName": "",
        }
      }
    />
    <NavigationContainer
      fixed={true}
      hideOnDesktop={false}
      right={false}
    >
      <BackButton
        disabled={false}
        onClick={[Function]}
        type="button"
      />
      <NextButton
        disabled={true}
        loading={false}
        type="submit"
      >
        <FormattedMessage
          defaultMessage="Save"
          id="settings.save"
          values={Object {}}
        />
      </NextButton>
    </NavigationContainer>
  </styled.form>
</ContextProvider>
`;
